op-node/node/node.go

```diff
@@ -319,13 +350,53 @@ func (n *OpNode) initMetricsServer(ctx context.Context, cfg *Config) error {
 	}
 	n.log.Info("starting metrics server", "addr", cfg.Metrics.ListenAddr, "port", cfg.Metrics.ListenPort)
 	go func() {
-		if err := n.metrics.Serve(ctx, cfg.Metrics.ListenAddr, cfg.Metrics.ListenPort); err != nil {
+		if err := n.metrics.Serve(n.postResourcesCtx, cfg.Metrics.ListenAddr, cfg.Metrics.ListenPort); err != nil {
 			log.Crit("error starting metrics server", "err", err)
 		}
 	}()
 	return nil
 }
 
+func (n *OpNode) initHeartbeat(_ context.Context, cfg *Config) error {
```

Comment:
This function always returns nil, can just remove the return value.

Reply:
The `go func` in there is a carry-over from the previous heartbeat setup, and I don't want to refactor the whole heartbeat service to `io.Closer` now, but we'll need the error-return back once the service doesn't drop the unhandled errors anymore. For now I'll simplify the code though, fair point.